distribution/ddf-common/src/main/resources/etc/shell.init.script

```diff
@@ -32,27 +32,38 @@ log:list = { log:get ALL } ;
 service:get = { $.context getService ($.context getServiceReference $args) };
 
 env = { shell:set $args }
-more = { shell:less $args }
+more = { shell:less -F $args }
 
-prompt = "[1m${USER}[0m@${APPLICATION}(${SUBSHELL})> "
+\#prompt = { "[1m${USER}[0m@${APPLICATION}(${SUBSHELL})> "?};
 
 __load_class = {
   (($.reader class) classLoader) loadClass $1
 }
 
-# On 256 colors terminal, add a right prompt
-max_colors = ($.jline.terminal getNumericCapability max_colors)
-if { %(max_colors >= 256) } {
-  __rprompt_formatter = (((__load_class java.text.SimpleDateFormat) getConstructor (__load_class java.lang.String)) newInstance \'$'\u001B\[90m'\'HH:mm:ss)
-  __date_class = (__load_class java.util.Date)
-  \#rprompt = { $__rprompt_formatter format ($__date_class newInstance) }
-}
+// make sure that we catch exceptions
+// as they do occur if the session is headless / non-interactive
+jlineReader = $.reader
+if { %(jlineReader != null) } {
+
+  # On 256 colors terminal, add a right prompt
+  max_colors = ($.jline.terminal getNumericCapability max_colors)
+  if { %(max_colors >= 256) } {
+    __rprompt_formatter = (((__load_class java.text.SimpleDateFormat) getConstructor (__load_class java.lang.String)) newInstance \'$'\u001B\[90m'\'HH:mm:ss)
+    __date_class = (__load_class java.util.Date)
+// Do not use right prompt by default
```

Comment:
[KARAF-5276](https://issues.apache.org/jira/browse/KARAF-5276) Do not use right prompt by default